### PR TITLE
GeoJSON: Enhance with generic properties

### PR DESCRIPTION
**Pull request by atd-schubert**
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: #20103
- [ ] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

This enhancement uses an optional generic type for feature properties. This way we don't have to update anything else, because nothing has changed as long as you don't use the second generic type. If you use the second generic type, you are able to specify your properties in detail by another type.

### CODE REVIEW

**Comment by atd-schubert** [comment]
At the moment I am the maintainer of @yaga/generic-geojson. With that typing-module you have the possibility to use types for your properties. The original specification just allows any type of property, so in my opinion the `any` type is correct.

But since typescript 2.3, you are able to use generics with a default, so an optional generic.

This way, I think, we can match both:
* The correct implementation of the specification
* The ability to specify the type of the properties

I would mark my library as deprecated on npm, after merging this PR.

**Comment by atd-schubert** [comment]
FYI: I just fixed the esri-leaflet Type definition in #20889

File: types/geojson/index.d.ts [hunk 1]
```diff
@@ -31,82 +32,82 @@ export interface DirectGeometryObject extends GeoJsonObject {
...
 /***
-* https://tools.ietf.org/html/rfc7946#section-3.2
-*/
-export interface Feature<T extends GeometryObject> extends GeoJsonObject {
-    type: 'Feature';
-    geometry: T;
-    properties: any;
+ * https://tools.ietf.org/html/rfc7946#section-3.2
+ */
+export interface Feature<G extends GeometryObject, P = any> extends GeoJsonObject {
```
Comment (atd-schubert) [inline types/geojson/index.d.ts]: Here is the actual change!

**Comment by atd-schubert** [comment]
@Cobster: If there is an interest, I can add myself to the list of authors for reviewing changes. I am a [OSGeo Charter Member](https://wiki.osgeo.org/wiki/User:Atd) and up-to-date with the spatial standards...

**Comment by andy-ms** [comment]
@atd-schubert Go ahead.

**Comment by atd-schubert** [comment]
This PR was more an idea after the discussion in #20103.

Here something about the background:

The structure of a GeoJSON is quite fixed and well defined in its type-definition on DefinitelyTyped. The only "open" property is the `properties` property of a GeoJSON feature. It represents the data you connect to your geometry. So in my opinion it is quite useful to specify your type-definition of your connected data.

The specification of GeoJSON just defines that you are able to apply any type of data to the `properties`property, so the definition at the moment is correct, in my opinion...

In addition I build a library `@yaga/generic-geojson` wich gives you the ability to specify the properties in detail and keep your object to be still a valid GeoJSON.

At the end I think it would be a good feature with an advantage, but there is also another possibility to realize this if you want to have this advantage.

**Comment by andy-ms** [comment]
OK, it sounds like the maintainers of the other packages also want something like this. You'll have to add a `// TypeScript Version: 2.3` comment to the dependent packages that are currently failing in travis.

**Comment by atd-schubert** [comment]
I have already add it in commit `df97fa1`, but I will take a look at it and fix the others...

**Comment by RyanCavanaugh** [comment]
@atd-schubert Please fix the failures indicated in the Travis CI log.

**Comment by atd-schubert** [comment]
@RyanCavanaugh what is the reason for the failure? I run the tests after the travis results, that I don’t really understand, on my computer and haven’t got any errors...

**Comment by atd-schubert** [comment]
@RyanCavanaugh what is the reason for the failure? I run the tests after the travis results, that I don’t really understand, on my computer and haven’t got any errors...

**Comment by andy-ms** [comment]
@RyanCavanaugh's mind has been taken over by a robot.
Could you rebase this PR against master? #21156 is needed to fix tests.

**Comment by andy-ms** [comment]
Thanks!

**Pull request merged**